my_methods.rb

```diff
@@ -0,0 +1,199 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/MethodLength
+# rubocop: disable Style/For
+# rubocop: disable Metrics/BlockNesting
+# rubocop: disable Metrics/CyclomaticComplexity
+# rubocop: disable Metrics/PerceivedComplexity
+# rubocop: disable Style/IfInsideElse
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    for item in array
+      yield item
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    for i in (0...array.length)
+      yield array[i], i
+    end
+  end
```

Comment:
The `my_each_with_index` method for hash values does not give the correct output.

```
h = {foo: 0, bar: 1, baz: 2}

#each_with_index

hash = Hash.new
h.each_with_index { |item, index|
  hash[item] = index
}

print hash  #=> {[:foo, 0]=>0, [:bar, 1]=>1, [:baz, 2]=>2}

#my_each_with_index

hash = Hash.new
h.my_each_with_index { |item, index|
  hash[item] = index
}

print hash #=> {0=>0, 1=>1, 2=>2}
```